### DISCUSSION TITLE
Using HTTP / HTTPS proxy to open website through Selenide

### DISCUSSION THREAD

**Discussion by Slavian2** [selenide/selenide#2061] [Q&A]
Hello!

Please advise how to properly use proxy servers in selenide.
I'd like to use a proxy server to open a website. I tried different approaches:

1) With / without Configuration.proxy*
    Configuration.proxyEnabled = true
    Configuration.proxyHost = "1.1.1.1"
    Configuration.proxyPort = 1111
2) open with BASIC authtype:

    Selenide.open("https://www.aaa.com/", AuthenticationType.BASIC, BasicAuthCredentials("1.1.1.1","login:pass"))
3) open with proxy server, login, pass:

        Selenide.open("https://www.aaa.com/", "1.1.1.1", "login","pass")
4) just open with Configuration;

Still no luck :-( I've googled across the web but with no luck; no real examples with proxy usage; documentation is too high level and still does now work:

https://selenide.org/javadoc/current/com/codeborne/selenide/[Selenide](https://selenide.org/javadoc/current/com/codeborne/selenide/Selenide.html#open(java.lang.String,java.lang.String,java.lang.String,java.lang.String)).html#open(java.lang.String,java.lang.String,java.lang.String,java.lang.String)

Please tell me which way I should go to understand how to use a custom proxy server (HTTP / HTTPS) in Selenide;

**Comment by asolntsev** [comment]
@Slavian2 Yes, you need to do it differently. :) 
Settings `Configuration.proxy*` say Selenide to start its own embedded proxy server. Right in the test. 

But you need to use some external proxy server, right?
Then you need method `setProxy`:
```java
WebDriverRunner.setProxy(seleniumProxy);
```

The full code would be similar to this:
```java
  import org.openqa.selenium;
  import com.codeborne.selenide.WebDriverRunner;

  Proxy proxy = new Proxy();
  proxy.setProxyType(Proxy.ProxyType.MANUAL);
  proxy.setHttpProxy("1.1.1.1:1111");
  proxy.setSslProxy("1.1.1.1:1111");

  WebDriverRunner.setProxy(seleniumProxy);
```

**Reply by Slavian2** [reply]
>   WebDriverRunner.setProxy(seleniumProxy);
I believe it should be 'proxy' variable here instead of 'seleniumProxy'?

How can add login / password for external proxy?

Can I use Selenide as usual after this part of code?

**Reply by asolntsev** [reply]
I am not sure, I never had to use it. 
Probably you can add username/password into proxy string:

```java
Proxy proxy = new Proxy();
  proxy.setProxyType(Proxy.ProxyType.MANUAL);
  proxy.setHttpProxy("http://login:password@1.1.1.1:1111");
  proxy.setSslProxy("https://login:password@1.1.1.1:1111");

  WebDriverRunner.setProxy(seleniumProxy);
```
